src/main/java/org/broadinstitute/consent/http/service/MetricsService.java

```diff
@@ -160,4 +172,33 @@ public String getHeaderRow(Type type) {
     }
     return election;
   }
+
+  public DatasetMetrics generateDatasetMetrics(Integer datasetId) {
+
+    DatasetMetrics metrics = new DatasetMetrics();
+
+    //get datasetDTO with properties and data use restrictions
+    Set<DatasetDTO> datasets = dataSetDAO.findDatasetDTOWithPropertiesByDatasetId(datasetId);
+    if (datasets == null || datasets.isEmpty()) {
+       throw new NotFoundException("Dataset with specified ID does not exist.");
+    }
+
+    //find all dars, then filter by datasetID because filtering cannot occur in DAO call because
+    //all dar data is saved in one column and not distinguishable until handled by the mapper
+    List<DataAccessRequest> dars = dataAccessRequestDAO.findAllDataAccessRequests();
```

Comment:
I can see the response getting pretty large if a dataset is used in a lot of DARs (the DARs are pretty big as a data struct and the aforementioned scenario becomes more likely as time passes). It might be a good idea to simplify the DAR data to only include properties needed for the metrics page.

There's probably a couple different ways to approach this. Off the top of my head you could make a DAO that returns only some of the DAR properties, or you can just fetchAll as normal and just pick out the data you need. Feel free to come up with a method that best tackles this scenario.

Reply:
Okay, good idea, I think I'll look at the pre-existing dar DAO calls to see if there's anything that I could use to get the info we need according to the mockups, and if not I'll make a new DAO call

Reply:
You can select/query a [specific field in the json](https://github.com/DataBiosphere/consent/blob/develop/src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java#L48) like so: `(data #>> '{}')::jsonb->>'sortDate'`. See also the postgres docs for some additional examples: https://www.postgresql.org/docs/11/functions-json.html

Reply:
Thanks greg I also found this to be very helpful: https://www.compose.com/articles/faster-operations-with-the-jsonb-data-type-in-postgresql/ 
I just made a new DAO query getting only the essential fields, and added an 'if' section to the mapper to account for the varied structure of these dars. Here is an example of what one looks like now! 
"dars": [
    {
      "id": 352,
      "referenceId": "938e4a2b-8435-485c-b8a8-c12236116c9f",
      "data": {
        "investigator": "Gregory Rushton",
        "projectTitle": "Collaborator Testing",
        "nonTechRus": "Collaborator Testing",
        "darCode": "DAR-261"
      },
      "draft": false,
      "userId": 3359,
      "createDate": 1597776900498,
      "sortDate": 1601994565814,
      "submissionDate": 1601994565814,
      "updateDate": 1601994565814
    }
]